core/src/main/scala/com/databricks/labs/remorph/intermediate/plans.scala

```diff
@@ -6,7 +6,7 @@ package com.databricks.labs.remorph.intermediate
  * the [[Command]] type that is used to execute commands on the server. [[Plan]] is a union of Spark's LogicalPlan and
  * QueryPlan.
  */
-abstract class Plan[PlanType <: Plan[PlanType]] extends TreeNode[PlanType] {
+abstract class Plan[PlanType <: Plan[PlanType]]()(origin: Origin) extends TreeNode[PlanType]()(origin) {
```

Comment:
origin in 2nd parameter list is required to only compare relevant fields

Reply:
As per [the guidelines](https://github.com/databricks/scala-style-guide?tab=readme-ov-file#multi-param-list) we should avoid multiple parameter lists. Could we come up with a solution that doesn't involve adding these parameter lists?

Reply:
I agree that mpl _should_ be `avoided`, but that's the only official way I know to exclude origin from comparisons in both TreeNode and all derived classes. I'm happy to consider any suggestion.

Reply:
From the back of my head, I guess something along the lines of:

```scala
trait TreeNode {
  private [this] var _origin: Origin = Origin.empty

  def origin: Origin = _origin

  def withOrigin(updatedOrigin: Origin): this.type = {
    _origin = updatedOrigin
    this 
  }
}
```

should do the trick